src/components/Swap/SwapButton/useSwitchNetwork.tsx

```diff
@@ -0,0 +1,43 @@
+import { Trans } from '@lingui/macro'
+import { Currency } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import { Action } from 'components/ActionButton'
+import { switchChain } from 'hooks/connectWeb3/useWeb3React'
+import { Spinner } from 'icons'
+import { useCallback, useMemo, useState } from 'react'
+
+export default function useSwitchNetwork(inputCurrency?: Currency) {
+  const { chainId, connector } = useWeb3React()
+
+  const [isPending, setIsPending] = useState(false)
+  const onSwitchNetwork = useCallback(
+    async (desiredChainId: number) => {
+      setIsPending(true)
+      try {
+        await switchChain(connector, desiredChainId)
+        setIsPending(false)
+      } catch {
+        // if user cancels switch network request in-wallet
```

Comment:
that's not the only thing it's handling necessarily, so i'd probably remove the comment